dataactbroker/fileRoutes.py

```diff
@@ -130,6 +130,46 @@ def get_protected_files():
         file_manager = FileHandler(request, is_local=is_local, server_path=server_path)
         return file_manager.get_protected_files()
 
+    @app.route("/v1/check_current_page/", methods=["GET"])
+    @convert_to_submission_id
+    def check_submission(submission_id):
+        sess = GlobalDB.db().session
+        # /v1/reviewData/
+        reviewdata = sess.query(Job).filter(Job.submission_id == submission_id,
+                                            Job.file_type_id.in_([6, 7]), Job.job_status_id == 4)
+        if reviewdata.count() > 0:
+            return "5"
+
+        # /v1/validateCrossFile/
+        validatecrossfile = sess.query(Job).filter(Job.submission_id == submission_id,
+                                                   Job.file_type_id.in_([4, 5]), Job.job_type_id == 2,
+                                                   Job.number_of_errors == 0, Job.file_size.isnot(None))
+        if validatecrossfile.count() > 0:
+            return "3"
+
+        # /v1/generateEF/
+        generateef = sess.query(Job).filter(Job.submission_id == submission_id, Job.file_type_id.in_([1, 2, 3, 4, 5]),
+                                            Job.job_status_id == 2, Job.number_of_errors == 0,
+                                            Job.file_size.isnot(None))
+        if generateef.count() > 0:
+            return "4"
+
+        # /v1/validateData/
+        validatedata = sess.query(Job).filter(Job.submission_id == submission_id,
+                                              Job.file_type_id.in_([1, 2, 3]), Job.job_type_id == 2,
+                                              Job.number_of_errors != 0, Job.file_size.isnot(None))
+        if validatedata.count() > 0:
+            return "1"
+
+        # /v1/generateFiles/
+        generatefiles = sess.query(Job).filter(Job.submission_id == submission_id,
+                                               Job.file_type_id.in_([1, 2, 3]), Job.job_type_id == 2,
+                                               Job.number_of_errors == 0, Job.file_size.isnot(None))
+        if generatefiles.count() > 0:
+            return "2"
+
+        return "0"
```

Comment:
All broker routes should return JSON responses as opposed to a string. For example: https://github.com/fedspendingtransparency/data-act-broker-backend/blob/development/dataactbroker/fileRoutes.py#L57-L61

Also, returning numbers isn't very descriptive. A string correlating to the page's name may be more descriptive/useful.

Additionally, when using numbers, if we were to add any pages in the process (might not be likely, but still possible), the route would require more updates to reorder what numbers should be returned versus just adding an additional check with a return value representing the name of the new page(s).